### DISCUSSION TITLE
Firebase Storage - useStorageFile() - 404 (Not Found)

### DISCUSSION THREAD

**Discussion by asifaly** [vuejs/vuefire#1283] [Questions and Help] [answered]
Hi, 

I am following the docs [here](https://vuefire.vuejs.org/guide/storage.html#uploading-files) to use firebase storage with vuefire.
In the example provided. as storage reference is set and this storage ref is passed to `useStorageFile` composable. The upload works fine with the provided example.

However, this seems to be causing a 404 not found error in the console.

<img width="1359" alt="image" src="https://user-images.githubusercontent.com/6282968/214013933-a703b6fa-21e8-476a-b3a3-c0042111d7f5.png">

Most of the time, you would know the filename only after upload by user and thus the storage reference too. How do you prevent this 404 error ?

**Comment by posva** [comment, marked as answer]
I think we might need to add an option to skip (initial) fetching until `upload()` is called. To avoid the network failed request (that's the error you see) you would have to upload using the firebase/storage api and only use `useStorageFile()` for files that already exist.

**Reply by asifaly** [reply]
@posva - so shall i maybe open this as an issue, for you to look at later to maybe add an option to skip initial fetching.

**Reply by posva** [reply]
no need, I already added to my todo list, thanks

**Reply by posva** [reply]
note maintainers can transform discussions into issues so this is never necessary, it's just one button away 😄

**Reply by BenJackGill** [reply]
I am getting the same error. @posva Was a fix for this released? If so, how do we set the option to skip fetching until `upload()` is called?

**Reply by posva** [reply]
This is tracked but not implemented yet

**Reply by paulwer** [reply]
any updates?

**Reply by szulcus** [reply]
Any updates?

**Reply by tomastrivino** [reply]
+1
